src/sagemaker/utils.py

```diff
@@ -749,32 +749,57 @@ def update_container_with_inference_params(
-        framework is not None
-        and framework_version is not None
-        and nearest_model_name is not None
-        and data_input_configuration is not None
-    ):
+    if framework is not None and framework_version is not None and nearest_model_name is not None:
         if container_list is not None:
             for obj in container_list:
-                obj.update(
-                    {
-                        "Framework": framework,
-                        "FrameworkVersion": framework_version,
-                        "NearestModelName": nearest_model_name,
-                        "ModelInput": {
-                            "DataInputConfig": data_input_configuration,
-                        },
-                    }
+                construct_container_object(
+                    obj, data_input_configuration, framework, framework_version, nearest_model_name
                 )
         if container_obj is not None:
-            container_obj.update(
-                {
-                    "Framework": framework,
-                    "FrameworkVersion": framework_version,
-                    "NearestModelName": nearest_model_name,
-                    "ModelInput": {
-                        "DataInputConfig": data_input_configuration,
-                    },
-                }
+            construct_container_object(
+                container_obj,
+                data_input_configuration,
+                framework,
+                framework_version,
+                nearest_model_name,
             )
+
+
+def construct_container_object(
+    obj, data_input_configuration, framework, framework_version, nearest_model_name
+):
+    """Function to construct container object.
+
+    Args:
+        framework (str): Machine learning framework of the model package container image
+                (default: None).
+        framework_version (str): Framework version of the Model Package Container Image
+            (default: None).
+        nearest_model_name (str): Name of a pre-trained machine learning benchmarked by
+            Amazon SageMaker Inference Recommender (default: None).
+        data_input_configuration (str): Input object for the model (default: None).
+        container_obj (dict): object to be updated.
+        container_list (list): list to be updated.
+
+    Returns:
+        dict: container object
+    """
+    if data_input_configuration is not None:
+        obj.update(
+            {
+                "Framework": framework,
+                "FrameworkVersion": framework_version,
+                "NearestModelName": nearest_model_name,
+                "ModelInput": {
+                    "DataInputConfig": data_input_configuration,
+                },
+            }
+        )
+    else:
+        obj.update(
+            {
+                "Framework": framework,
+                "FrameworkVersion": framework_version,
+                "NearestModelName": nearest_model_name,
+            }
+        )
```

Comment:
nit : Can the obj be partially updated also? Then we can refactor the code saying update the 3 fields(Framework, FrameworkVersion,  NearestModelName) first and then if `data_input_configuration` is not none then update `ModelInput`

Reply:
Yes, this can be refactored. Thank you for pointing this out!